portfolio/src/main/webapp/script.js

```diff
@@ -78,43 +78,63 @@ function openTab(evt, sectionName) {
 }
 
 /**
- * Fetch function from servlet
+ * Get all comments from the server
  */
-async function getWelcomeData() {
+async function getComments() {
+    // Clear old comments
+    const commentsEl = document.getElementById('comment_list');
+    // for some reason this wasn't working $("comment_list").empty();
+    emptyElement(commentsEl);
+
+    // Determine display preferences
+    const numComments = document.getElementById('comment_count').value;
+    const sortType = document.getElementById('comment_sorting').value;
+
     // Get comments from the server
-    const response = await fetch('/data');
-    let comments = await response.json();
+    const queryString = '?count=' + numComments + '&sort=' + sortType;
+    const response = await fetch('/data' + queryString);
+    const comments = await response.json();
 
     // Display the comments
-    const commentsEl = document.getElementById('comment_list');
     comments.forEach((comment) => {
         commentsEl.appendChild(createCommentElement(comment));
     });
 }
 
+/**
+ * Deletes all children of a certain element
+ */
+function emptyElement(element) {
+    while (element.lastChild) {
+        element.removeChild(element.lastChild);
+    }
+}
+
 /**
  * Creates one comment element
  */
 function createCommentElement(comment) {
     const commentElement = document.createElement('li');
-    commentElement.className = 'comment';
+    commentElement.classList.add('comment');
```

Comment:
nit: let's be consistent and use double quotes throughout (e.g. "comment", not 'comment')